function-service-impl/src/main/java/com/future/function/service/impl/feature/core/ActivityBlogServiceImpl.java

```diff
@@ -52,8 +53,9 @@ public ActivityBlog getActivityBlog(String activityBlogId) {
     String userId, String search, Pageable pageable
   ) {
 
-    return Optional.ofNullable(
-      activityBlogRepository.findAll(userId, search, pageable))
+    return Optional.ofNullable(userId)
+      .filter(ObjectId::isValid)
```

Comment:
is this necessary? because if it's not valid, it will return empty page right?

Reply:
If invalid, it will throw IllegalArgumentException.

The repository method is custom, and to find activity blogs by specified userId, we need to create `ObjectId` object (`new ObjectId(string_object)`). However, if the `string_object` does not conform to ObjectId's constructor parameter requirement (which should be hex string), it throws `IllegalArgumentException`. This will return error code 500 to client.

Rather than returning error code 500, I prefer to prevent such behavior, hence the code in line 57. If the condition is not fulfilled, meaning the userId is not of ObjectId required format, empty page will be returned.